src/frontend/src/expr/mod.rs

```diff
@@ -96,9 +98,16 @@ impl_expr_impl!(
     Subquery,
     TableFunction,
     WindowFunction,
-    UserDefinedFunction
+    UserDefinedFunction,
+    Parameter
 );
 
+impl Default for ExprImpl {
+    fn default() -> Self {
+        Self::literal_null(DataType::Varchar)
+    }
+}
+
```

Comment:
How much extra effort does it cost if we do not have `Default`? I have been trying to avoid it and uses `swap` / `replace` rather than `take`
https://github.com/risingwavelabs/risingwave/blob/4f34adeaa51c1260640c7de20c58fffccfb26024/src/frontend/src/expr/type_inference/cast.rs#L65-L68
https://github.com/risingwavelabs/risingwave/blob/4f34adeaa51c1260640c7de20c58fffccfb26024/src/frontend/src/expr/type_inference/func.rs#L365-L366

In most cases, the "default" value (no matter `null::varchar` here or `false` above) should never be read. It should be overwritten immediately. It is not a meaningful default value and misuse could cause troubles.

Reply:
Seems we don't need it in this PR. And I will avoid to use it by using swap / replace in the next PR of binding param.